internal/cli/streams/privatelink/create_test.go

```diff
@@ -0,0 +1,149 @@
+			fileContents: `
+				{
+					"region": "US_EAST_2",
+					"serviceEndpointId": "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace",
+					"dnsDomain": "test-namespace.servicebus.windows.net"
+				}
+			`,
+			wantErr: func(tt require.TestingT, err error, _ ...any) {
+				require.ErrorContains(tt, err, "provider missing")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+
+			if len(tc.fileContents) > 0 {
+				require.NoError(t, afero.WriteFile(fs, fileName, []byte(tc.fileContents), 0600))
+			}
+
+			createOpts := &CreateOpts{
+				fs:       fs,
+				filename: fileName,
+			}
+
+			tc.wantErr(t, createOpts.Run())
+		})
+	}
+
+	t.Run("should call the store create privateLink method with the correct parameters", func(t *testing.T) {
+		fs := afero.NewMemMapFs()
+		require.NoError(t, afero.WriteFile(fs, fileName, []byte(validPrivateLinkConfigFileContents), 0600))
+
+		ctrl := gomock.NewController(t)
+		mockStore := mocks.NewMockPrivateLinkCreator(ctrl)
+
+		createOpts := &CreateOpts{
+			store:    mockStore,
+			fs:       fs,
+			filename: fileName,
+		}
+
+		expectedProvider := "Azure"
+		expectedRegion := "US_EAST_2"
+		expectedServiceEndpointID := "/subscriptions/fd01adff-b37e-4693-8497-83ecf183a145/resourceGroups/test-rg/providers/Microsoft.EventHub/namespaces/test-namespace"
+		expectedDNSDomain := "test-namespace.servicebus.windows.net"
+
+		expected := atlasv2.NewStreamsPrivateLinkConnection()
+		expected.Provider = &expectedProvider
+		expected.Region = &expectedRegion
+		expected.ServiceEndpointId = &expectedServiceEndpointID
+		expected.DnsDomain = &expectedDNSDomain
+
+		mockStore.
+			EXPECT().
+			CreatePrivateLinkEndpoint(gomock.Eq(createOpts.ConfigProjectID()), gomock.Eq(expected)).
+			Times(1)
+
+		require.NoError(t, createOpts.Run())
+	})
+
+	t.Run("should print the result", func(t *testing.T) {
+		fs := afero.NewMemMapFs()
+		require.NoError(t, afero.WriteFile(fs, fileName, []byte(validPrivateLinkConfigFileContents), 0600))
+
+		ctrl := gomock.NewController(t)
+		mockStore := mocks.NewMockPrivateLinkCreator(ctrl)
+
+		buf := new(bytes.Buffer)
+		createOpts := &CreateOpts{
+			store:    mockStore,
+			fs:       fs,
+			filename: fileName,
+			OutputOpts: cli.OutputOpts{
+				Template:  createTemplate,
+				OutWriter: buf,
+			},
+		}
+
+		expectedInterfaceEndpointID := "vpc-1234567890abcdef0"
+
+		expected := atlasv2.NewStreamsPrivateLinkConnection()
+		expected.InterfaceEndpointId = &expectedInterfaceEndpointID
+
+		mockStore.
+			EXPECT().
+			// This test does not assert the parameters passed to the store method
+			CreatePrivateLinkEndpoint(gomock.Any(), gomock.Any()).
+			Return(expected, nil).
+			Times(1)
+
+		require.NoError(t, createOpts.Run())
+		assert.Equal(t, "Atlas Stream Processing PrivateLink endpoint "+expectedInterfaceEndpointID+" created.\n", buf.String())
```

Comment:
this can for sure be part of the test matrix

Reply:
I am running into issues bringing this up into the test matrix function because the mock store does not get called so the assertions (even though they are weak and I don't care about them) fail, but I need them to force a store return value that I can consistently assert against for this string where this comment is. Adding conditionals to my test matrix to have different branches within my test defeats the purpose of a test matrix in my opinion so I think leaving this as a separate test makes the most sense.

Reply:
yeah if too hard no worries but was worth flagging that output check could be part of a matrix